app/models/concerns/revision_count.rb

```diff
@@ -0,0 +1,13 @@
+require 'active_support/concern'
+
+module RevisionCount
+  extend ActiveSupport::Concern
+
+  included do
+    after_update :increment_revision
+  end
+
+  def increment_revision
+    conference.increment!(:revision)
```

Comment:
Ok, so the problem is not with duck typing, which works fine!

The problem is with `#increment!`. Let's first consider that `#increment_revision` gets called on `after_update`. With that in mind, the problem is that `#increment!` ALSO triggers callbacks, so it calls `after_update` and thus `#increment_revision` AGAIN. And again. And again. Which is why you get "stack level too deep".

So to update the `revision`, we need to do it using a method that will NOT trigger callbacks.